### PR TITLE
Sleep Sort in Java

### PR DESCRIPTION
**Pull request by sakshikhachane**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added sleep sort in java

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
 #2018

### CODE REVIEW

**Comment by sakshikhachane** [comment]
@anushkrishnav Please review the PR.

**Comment by shraddhavp** [comment]
@sakshikhachane  the code has static input already given,Can you change the code to take input from t he user dynamically?

**Comment by sakshikhachane** [comment]
@shraddhavp Changes done to add the input dynamically.

**Comment by shraddhavp** [comment]
@sakshikhachane  also add the time and space complexity ,then it is ready to merge

**Comment by sakshikhachane** [comment]
@shraddhavp Did the changes.

**Comment by shraddhavp** [comment]
LGTM! @iamrajiv  please once review it

File: Java/sort/SleepSort.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import java.util.Scanner;
```
Comment (HarshCasper) [inline Java/sort/SleepSort.java:6]: Add a Beginner Pitch about your Code

**Review by iamrajiv — APPROVED** [review]

File: Java/sort/SleepSort.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+                        countDone.await();
+                        Thread.sleep(num);
+                        System.out.println(num);
+
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }).start();
```
Comment (HarshCasper) [inline Java/sort/SleepSort.java]: Can you add proper comments?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny changes required. LGTM 👍

File: Java/sort/SleepSort.java (hunk 1 above)
Comment (sakshikhachane) [reply Java/sort/SleepSort.java:6]: @HarshCasper Done with the changes.

**Comment by sakshikhachane** [comment]
@HarshCasper Please review this now. I have done all the changes.

**Pull request merged**